src/Core/Context.idr

```diff
@@ -2143,6 +2143,14 @@ getWorkingDir
               | Nothing => throw (InternalError "Can't get current directory")
          pure d
 
+export
+withCtxt : {auto c : Ref Ctxt Defs} -> Core a -> Core a
+withCtxt = wrapRef Ctxt resetCtxt
+  where
+    resetCtxt : Defs -> Core ()
```

Comment:
Sorry that I am writing this only now: I think we should `branch` the context here _and_ keep the new `timings`.
If we don't branch the context, some changes, that might have been made during the callback, will *persist* after the reversal to the original version of `Defs`. That happens because definitions are stored a mutable array - its state will persist across `get/put` calls.
Here is the array I am talking about: https://github.com/idris-lang/Idris2/blob/d6583048d8682b28bb9d611ca44c6dac957cd6b8/src/Core/Context.idr#L330

Am I right about this, @gallais ?

Reply:
For reference, this is how Edwin handles backtracking:
https://github.com/idris-lang/Idris2/blob/71abc8e33b17adba61173f3ce1c7564a1d9ce4dc/src/TTImp/Elab/Check.idr#L465-L484